pkg/rpcclient/rolemgmt/roles_test.go

```diff
@@ -149,3 +152,45 @@ func TestDesignateAsRoleTransaction(t *testing.T) {
 		require.Equal(t, ta.tx, tx)
 	}
 }
+
+func TestDesignationEvent_FromStackItem(t *testing.T) {
+	tests := []struct {
+		name        string
+		items       []stackitem.Item
+		expectedErr error
+		expected    DesignationEvent
+	}{
+		{
+			name:        "role overflow",
+			items:       []stackitem.Item{stackitem.NewBigInteger(big.NewInt(256)), stackitem.NewBigInteger(big.NewInt(1))},
+			expectedErr: fmt.Errorf("role overflow: %d > %d", 256, math.MaxUint8),
+		},
+		{
+			name:        "blockIndex overflow",
+			items:       []stackitem.Item{stackitem.NewBigInteger(big.NewInt(4)), stackitem.NewBigInteger(big.NewInt(int64(math.MaxUint32) + 1))},
+			expectedErr: fmt.Errorf("blockIndex overflow: %d > %d", int64(math.MaxUint32)+1, math.MaxUint32),
+		},
+		{
+			name:  "success",
+			items: []stackitem.Item{stackitem.NewBigInteger(big.NewInt(4)), stackitem.NewBigInteger(big.NewInt(123))},
+			expected: DesignationEvent{
+				Role:       4,
+				BlockIndex: 123,
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			item := stackitem.NewArray(tc.items)
+			var event DesignationEvent
+			err := event.FromStackItem(item)
```

Comment:
```suggestion
			var event DesignationEvent
			err := event.FromStackItem(stackitem.NewArray(tc.items))
```

Code style nitpicking: `item` is used in a single place and short enough to be inlined.